pyccl/nl_pt/lpt.py

```diff
@@ -0,0 +1,492 @@
+        Pdmd2 = 0.5*self.lpt_table[:, :, 4]
+        Pdms2 = 0.25*self.lpt_table[:, :, 7]
+        Pdmo3 = 0.25 * self.lpt_table[:, :, 11]
+
+        if self.pk_bk is not None:
+            Pdmk2 = 0.5*self.pk_bk * (self.k_s**2)[None, :]
+        else:
+            Pdmdm = self.lpt_table[:, :, 1]
+
+            Pdmk2 = 0.5*Pdmdm * (self.k_s**2)[None, :]
+
+        pgm += (b2[:, None] * Pdmd2 +
+                bs[:, None] * Pdms2 +
+                b3nl[:, None] * Pdmo3 +
+                bk2[:, None] * Pdmk2)
+
+        return pgm*self.exp_cutoff
+
+    def _get_pmm(self):
+        """ Get the one-loop matter power spectrum.
+
+        Returns:
+            array: 2D array of shape `(N_a, N_k)`, where `N_k` \
+                is the size of this object's `k_s` attribute, and \
+                `N_a` is the size of the object's `a_s` attribute.
+        """
+        self._check_init()
+
+        pk = self.lpt_table[:, :, 1]
+
+        return pk*self.exp_cutoff
+
+    def get_biased_pk2d(self, tracer1, *, tracer2=None,
+                        extrap_order_lok=1, extrap_order_hik=2):
+        """Returns a :class:`~pyccl.pk2d.Pk2D` object containing
+        the PT power spectrum for two quantities defined by
+        two :class:`~pyccl.nl_pt.tracers.PTTracer` objects.
+
+        .. note:: The full non-linear model for the cross-correlation
+                  between number counts and intrinsic alignments is
+                  still work in progress in FastPT. As a workaround
+                  CCL assumes a non-linear treatment of IAs, but only
+                  linearly biased number counts.
+
+        Args:
+            tracer1 (:class:`~pyccl.nl_pt.tracers.PTTracer`): the first
+                tracer being correlated.
+            tracer2 (:class:`~pyccl.nl_pt.tracers.PTTracer`): the second
+                tracer being correlated. If ``None``, the auto-correlation
+                of the first tracer will be returned.
+            extrap_order_lok (:obj:`int`): extrapolation order to be used on
+                k-values below the minimum of the splines. See
+                :class:`~pyccl.pk2d.Pk2D`.
+            extrap_order_hik (:obj:`int`): extrapolation order to be used on
+                k-values above the maximum of the splines. See
+                :class:`~pyccl.pk2d.Pk2D`.
+
+        Returns:
+            :class:`~pyccl.pk2d.Pk2D`: PT power spectrum.
+        """
+        if tracer2 is None:
+            tracer2 = tracer1
+
+        t1 = tracer1.type
+        t2 = tracer2.type
+
+        if ((t1 == 'NC') or (t2 == 'NC')) and (not self.with_NC):
+            raise ValueError("Can't use number counts tracer in "
+                             "LagrangianPTCalculator with 'with_NC=False'")
+
+        if t1 == 'NC':
+            if t2 == 'NC':
+                pk = self._get_pgg(tracer1, tracer2)
+            else:  # Must be matter
+                pk = self._get_pgm(tracer1)
+        else:  # Must be matter
```

Comment:
Here it could be IAs, right? I think we need to check or throw an exception earlier on if any of the tracers is IAs.

Reply:
Added.